src/main/java/app/controller/linAlg/Vector.java

```diff
@@ -68,7 +70,15 @@ public Vector rotate(double degrees)
     public double getAngle()
     {
         // + 90 to set the angle to north, negate angle to make it clockwise
-        return (-Math.toDegrees(Math.atan2(y,x)) + 360.0 + 90.0) % 360.0;
+        // Rounded to 1 decimal place
+        double angle = (-Math.toDegrees(Math.atan2(y,x)) + 360.0 + 90.0) % 360.0;
+        double roundedAngle = (double) Math.round(angle * 10) / 10;
+
+        if(roundedAngle == 360.0)
```

Comment:
don't you need to take into account that doubles aren't precise?

for example:

if( Math.abs(roundedAngle-360.0) < 0.00001 )
```suggestion
        if(Math.abs(roundedAngle-360.0) < 0.00001 )```